src/main/java/com/exadel/discount/entity/Category.java

```diff
@@ -0,0 +1,40 @@
+package com.exadel.discount.entity;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
+import java.util.List;
+import java.util.UUID;
+
+@Getter @Setter
+@EqualsAndHashCode
+@NoArgsConstructor
+@Entity
+@Table(name = "categories")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private UUID id;
+
+    private String name;
+
+    @OneToMany(cascade = CascadeType.ALL,
+    fetch = FetchType.EAGER,
+    mappedBy = "category")
+    private List<Discount> discount;
+
+    public Category(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public String toString() {
+        return "Category{" +
+                "id=" + id +
+                ", name='" + name + '\'' +
+                '}';
+    }
```

Comment:
I don't understand the aim of overriding toString() by hand. We can use Lombok's @ToString(exclude ={"someField", ...}). And the same for hachCode. You could write @Data and then @ToString(exclude = {"discounts"}), @EqualsAndHashCode (exclude = {"discounts"). It will brief than @Getter @Setter.

Reply:
+